libraries/AP_NavEKF2/AP_NavEKF2_Measurements.cpp

```diff
@@ -521,6 +521,14 @@ void NavEKF2_core::readGpsData()
                 gpsNotAvailable = false;
             }
 
+            // if we are initialising earth field from tables and we
```

Comment:
I would prefer that we don't write to the EKF states from inside the GPS measurement function. It would be better to do this check from a function responsible for controlling  mag and resets.